thing-url-adapter.js

```diff
@@ -799,10 +838,49 @@ function loadThingURLAdapter(addonManager) {
       adapter.pollInterval = config.pollInterval * 1000;
     }
 
-    for (const url of config.urls) {
-      adapter.loadThing(url);
+    let modified = false;
+    for (const entry in config.urls) {
+      if (config.urls.hasOwnProperty(entry)) {
+        if (typeof config.urls[entry] === 'string') {
+          config.urls[entry] =
+              {href: config.urls[entry], authentication: {method: 'none'}};
+          modified = true;
+        }
+      }
+    }
+    if (modified) {
+      db.saveConfig(config);
     }
 
+    for (const url of config.urls) {
+      if ('authentication' in url) {
+        // remove http(s) from url
+        let url_stub = '';
+        if (url.href.includes('http://')) {
+          url_stub = url.href.substr(7);
+        }
+        if (url.href.includes('https://')) {
+          url_stub = url.href.substr(8);
+        }
+        switch (url.authentication.method) {
+          case 'jwt':
+            adapter.authData[url_stub] = ['jwt', url.authentication.token];
```

Comment:
Setting the array item to the authentication object would be clearer, and would prevent the zero index further up.

Reply:
I'm not following the zero index part. Where is this?

Reply:
Line 542.

Reply:
Sorry, still not seeing the issue. 
It won't get to 542 unless an entry is found in auth. I'm thinking using an object could have the same issue. If this is a problem here it will be a problem at 272 as well. 

But I can see the elegance of using the part of authentication object. 
Just so I don't endup reworking this again I plan on using:
{method: 'jwt", token: "etc"} or do you have another preference?

Reply:
Changed